### PR TITLE
[Windows] Fix for the issue with Label text color when resetting from FormattedText.

### LINKED ISSUE #24516: Rendering issue in WinUI when setting Label.FormattedText
### Description

When setting a formatted text in Label works as expected, but when setting the Text property again, part of the format is visible in the label in WinUI.
It works ok in Android.

### Steps to Reproduce

1. Open attached project [MauiApp4.zip](https://github.com/user-attachments/files/16802443/MauiApp4.zip)
2. Run the application on Windows
3. Press "Click me" button. (The "Hello, World!" text is highlighted correctly)
4. Press "Click me" again. The format is not removed.

image

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.40 SR5

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

_No response_

### Did you find any workaround?

No

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by Tamilarasan-Paranthaman**
### Root Cause of the issue 

When changing the label text to FormattedText with text color, and then switching back to normal text, the text retains the formatted text color. This is due to the TextHighlighter applied in the UpdateInlines method of FormattedStringExtensions, which is not cleared when reverting to normal text.

### Description of Change
 
I cleared the TextHighlighters in the UpdateText method of the TextBlockExtensions class before setting the normal text, which resolves the issue and works fine.

### Issues Fixed

Fixes #24516

### Tested the behaviour in the following platforms

- [x] Android
- [x] Windows
- [x] iOS
- [x] Mac

### Screenshot

| Before Issue Fix | After Issue Fix |
|----------|----------|
| <video width="270" height="600"  src="https://github.com/user-attachments/assets/5d808f3f-1a78-42e2-a03d-94617ae7ed43"> | <video width="300" height="600"  src="https://github.com/user-attachments/assets/99f27829-4530-4cba-bc44-cdc6cb4294ad"> |

### CODE REVIEW

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/src/Core/Label/Label.cs [hunk 1]
```diff
@@ -55,7 +55,9 @@ public partial class Label : View, IFontElement, ITextElement, ITextAlignmentEle
 				{
 					var formattedString = ((FormattedString)oldvalue);
 					var label = ((Label)bindable);
-
+#if WINDOWS
+					(label.Handler as LabelHandler)?.PlatformView?.TextHighlighters?.Clear();
```
[the remark this answers is not in the document]
Comment (Tamilarasan-Paranthaman) [reply src/Controls/src/Core/Label/Label.cs]: I have moved the fix into the TextBlockExtensions and ensured that the issue is resolved. Please review it and share if you have any concerns.

**Review by Foda — APPROVED** [review]

**Review by PureWeen — APPROVED** [review]

**Pull request merged**